### ISSUE TITLE
How to market a product for developers?

### ISSUE THREAD

**Issue by Loccolito** [why-not-here/because#2]
### TL;DR

1. Don't do marketing.
2. Don't hire marketers.
3. If it's too late for the 2., don't worry, either they'll leave on their own, or they'll learn to code.

### Apéro tips (for marketers)
You have to work for a dev tools company to get your ass kicked.

### Introduction

When you think of marketing, what comes to mind?

A cold email that starts with “Hello {first_name}”? An ad for Pipedrive in Google results after searching for Salesforce? A newbie who joins a Machine Learning subreddit to share the latest landing page for his “powered by AI” product (guilty)? Anything else?

The caricature is not so far from the truth…and the developers won't tell you otherwise. In fact, they won't tell you much because they don't like social interactions (they too won't be spared).

For a developer tools company, this traditional playbook is not just ineffective, it’s just embarrassing. Developers don’t just ignore these tactics, they mock them. As a marketer, I have to admit that the problem isn't them. We're the problem.

More in the next comment.

**Comment by Loccolito** [comment]
### The “anti-marketing” approach (that's still marketing)
Let’s face it: most marketing is built on wind and glitter. That might work for selling sneakers or energy drinks, but software for developers?
1. Let your product do the talking

Developers are doers. They want to get their hands dirty. Please marketers, stop with your positioning statement, value propositions or USP (unique selling proposition). All you're doing is confusing their understanding of your product. Just let them use your product. And for nothing in return. Some companies have done it perfectly. I'm thinking in particular of Postman’s free API testing platform and Docker’s open-source containerization tool. But as many, you're afraid your product isn't ready? Is it too early? Spoiler: it'll never be the right time. Test it with your audience as soon as possible. Their feedback will sometimes be hard. But they'll almost always be sympathetic, even if the form isn't there. Forgive them, their lives are full of rejected pull requests. They're no longer used to taking the short end of the stick. And good for you. Put your ego aside and listen to them ALL.

More in the next comment.

**Comment by Loccolito** [comment]
2. Documentation is marketing

To continue with marketers reflexes, you must stop spending too much time into your website or your landing page. You wasting your time for nothing. Have you ever seen a shop with a beautiful store front with nothing inside? No, so stop it now.

It’s not the call to action wording or the tagline of the hero that will drive your growth. Just be clear and conscis, few lines of text would do the job correctly. In most of cases, a Notion is enough to present your product or better your Readme Github repository!

No need to think about If you have to choose between Framer, Webflow or Strapi because everyone doesn’t care about that. Instead of that, the only place that you should spend time to writing content is for your technical documentation.

Bad documentation isn’t just a missed opportunity. It’s an insult. If your docs are a mess, developers will assume your tool is too. Comprehensive, well-organized, and developer-friendly documentation isn’t a bonus: it’s the bare minimum. Stripe and Twilio didn’t dominate their markets because of slick branding. They crushed it because their documentation makes developers look like geniuses.

Few weeks ago, I took the time to explore the Stripe API documentation. And what a claque! I’m not developer and my tech skills are near to the level of the sea but it was so easy to understand. A 7 years old child could easily follow the [step-by-step guide](https://docs.stripe.com/stripe-cli) to get started with the [Stripe CLI](https://github.com/stripe/stripe-cli). The documentation was so strategic for the Stripe team that they decided to create their own tool for creating custom documentation aka https://markdoc.dev/.

Anyway, please respect developers and provide them a clear and complete documentation. If you don’t trust me, you should trust Nicolas Dessaigne (Algolia co-founder):
<img width="895" alt="Doc-is-marketing" src="https://github.com/user-attachments/assets/cfce8b2e-5a31-4112-8705-632ab24751a2" />

**Comment by Loccolito** [comment]
3. Group strength: community-first

A strong developer community is one of the most powerful assets a dev tools startup can have. Developers trust other developers more than they’ll ever trust your marketing.

Developers are inherently collaborative. They rely on each other to solve problems, share knowledge, and validate tools. When you build a thriving community, you’re creating a space where devs can learn from one another, share use cases, and even advocate for your product without you lifting a finger. But beware, if your product is crap, they will not hesitate to knock it down.

You have to create spaces where they can connect, share, and collaborate. Look at Algolia’s Discord or [CircleCI’s community](https://discuss.circleci.com/). These are not marketing stunts, they’re ecosystems where developers solve real problems together. 

Magic formulas only exist in Disney movies, everyone knows the foundations of a community (even if very few actually apply them):

- Create value-driven spaces:
    - Host forums, Discord servers, or GitHub discussions where developers can ask questions, share projects, and troubleshoot issues.
    - Focus on making these spaces genuinely useful, not just another place to push your product.
- Recognize and empower advocates:
    - Identify early adopters and power users who love your tool.
    - Offer them perks: early access to features, exclusive swag, or recognition as community leaders.
- Be present and authentic:
    - Show up in the community yourself. Whether it’s answering questions or sharing updates, your involvement builds trust.
    - Avoid the hard sell. Developers value authenticity and transparency above all else.
- Organize events
    - Host webinars or local meetups. These events can foster deeper connections and inspire developers to share their experiences with your tool.

And avoid the ultimate trap room: don’t micromanage or overly commercialize your community. Nothing kills engagement faster than turning a developer forum into a marketing echo chamber.

Rest assured, we're not talking about a mission impossible. Some have perfectly understood this, and I'm thinking in particular of : 

- [HashiCorp’s community](https://www.hashicorp.com/community): offers forums, local meetups, and even "office hours" for developers to ask questions.
- [Kubernetes](https://kubernetes.slack.com/?redir=%2Fmessages%2FCNXNB0ZTN): their Slack and GitHub discussions are legendary for peer-to-peer problem-solving.
- [Twilio’s community](https://stackoverflow.com/collectives/twilio?_ga=2.61669996.1944049906.1651632140-1148610530.1651632140): developers share creative use cases, making it a treasure trove of inspiration and advocacy.

I don't need to give you 150 examples, as you know (well, not that well), the community is powerful. A power that works both ways. As powerful to lift you up as it is to bury you. Put aside your thirst for sales and your P&L and everything should work out perfectly.

Don’t forget about that: a thriving developer community doesn’t just support your product, it amplifies it. That’s influence you can’t buy.

More in the next comment.

**Comment by Loccolito** [comment]
4. Focus on developer pain points (and stop talking about your product)

![Image](https://github.com/user-attachments/assets/d3133960-8e9f-4614-a96d-410045fbe20c)

Let’s get something straight: developers don’t care about your product. They care about solving their own problems, as fast as possible, with as little friction as possible. If your marketing isn’t laser-focused on how your tool makes their life easier, you’re wasting everyone’s time, including your own.

Yet, most marketing teams get lost hyping their "AI-powered, next-gen cloud-native synergy engine bla bla bla" instead of answering the only question that matters: How does this help developers ship code faster, more efficiently, or with fewer bugs?

It's not about selling hot air with fancy words (I like you copywriters, but you're starting to get us all puffed up with your nauseating phrases). Stop talking about your features and listing them as if you had to go to the supermarket with your grandmother. Turn them into benefits for your potential users. I say it again, but shiny marketing won't work here. And yet, everyone keeps bringing out heroes that nobody understands. Come on, let's take a look at the fundamentals that nobody applies (me first).

1. Stop selling, start solving
    - Nobody wants to hear about your product’s "innovative features." They want to know how it fixes the daily annoyances they’re already dealing with.
    - Instead of bragging about a new API endpoint, show how it eliminates a tedious manual process.
2. Make performance the priority
    - If your product slows developers down, they will abandon it faster than you can say "conversion funnel."
    - Technical benchmarks, speed comparisons, and real-world load tests are far more persuasive than any slick marketing claim.
3. Show real use cases, not hypothetical scenarios
    - A developer debugging an issue at 2 AM isn’t looking for "theoretically improved workflow efficiency." He wants a step-by-step example showing how your tool fixes their exact issue.
    - Case studies that dive into real-world problems are significantly more valuable than vague whitepapers (I can hear a marketer in the background saying, “Yes, but that's a successful lead magnet for us.” As The Notorious would say “shut your ******* mouth”).
4. Make your product work with existing developer stack
    - No one wants to rip out their entire stack just to use your tool.
    - Show integrations, compatibility with popular frameworks, and minimal setup time.
5. Address skepticism head-on
    - Developers assume your tool won’t work until proven otherwise. Accept that challenge.
    - Provide interactive demos, transparent benchmarks, and clear comparisons with alternatives.

The key takeaway? Developers don’t want your product. They want solutions to their problems. If your marketing doesn’t clearly demonstrate that, you’re just adding noise to their already workload.

More in the next comment.

**Comment by Loccolito** [comment]
5. Build trust through transparency

![Image](https://github.com/user-attachments/assets/478376ef-9833-4fc8-8012-519ffd0f6f0f)

Let’s be clear, developers don’t believe your marketing. By default, they assume you’re exaggerating, overselling, or straight-up lying about your product. And honestly? They're often right.

Every DevTool company claims to be fast, scalable, and easy to use. The problem? Most of them are full of it:

- “Our API is simple!” → Until they see a 10-page quick start guide and an incoherent documentation.
- “Plug and play!” → Until they spend hours debugging dependencies.
- “Blazing fast!” → Until it crashes under real-world usage.

Developers have been burned too many times. They don’t trust what you say, they trust what they can verify. So instead of getting offended when they’re skeptical, accept the challenge and prove them wrong.

Here are a few thoughts:

**1. Kill the bullshit: show real benchmarks**

Developers have seen too many “blazing fast” tools that crumble under real-world usage. Too many “easy to integrate” platforms that require days of setup. Your job? Disarm their skepticism by:

- Showing benchmarks with raw data.
- Exposing the edge cases where your tool isn’t the best (and what you’re doing about it).
- Giving as much information as possible about the context of your tests.

If you try to sugarcoat weaknesses, they’ll find them anyway, and destroy you for hiding them.

If you don’t, they will assume you cherry-picked data or worse, made it up. And this is still all too often the case. We see you!

**2. Let them break your product**

Demos are great. Free trials are better. Cherry on the cake, leave them a completely free playground.

- If a dev can’t test your product without talking to sales, they’re gone.
- If they need to enter credit card details just to see if it works, they’ll find an alternative.

They don’t want polished marketing fluff. They want a playground where they can push your tool to its limits. If you believe in your product, let them try to break it.

**3. Make switching frictionless**

Your biggest competitor isn’t another DevTool startup, it’s inertia.

- If adopting your tool means ripping out half their existing stack, they won’t even consider it.
- If they need to sit through hours of onboarding calls, they’ll bounce.

Make migration stupidly easy by:

- Providing scripts and automate setup.
- Providing an open, well-documented API.
- Offer a working sandbox or instant demo.
- Give them real code snippets that they can copy-paste and run immediately.

If they have to dig through 10 pages of documentation just to understand how to make a request, you’ve already lost them. It’s easy: reduce friction everywhere!

**4. Own your weaknesses before devs expose them**

Every tool has flaws. The difference? Some companies admit it upfront, while others wait to get roasted on GitHub issues.

- If your product isn’t great at X, say it.
- If there’s a learning curve, acknowledge it.
- If competitors do something better, be transparent about it.

Developers don’t believe in corporate hype, but they do believe in what other devs say.

- If your tool is actually good, they’ll talk about it on GitHub, X, and Reddit.
- If it sucks, they’ll be even louder.

Developers don’t expect perfection. They expect honesty. And if they feel like you’re hiding something, they’ll dig until they find it, and then call you out publicly.

Invest in real case studies, community-driven content, and open discussions rather than marketing fluff. Nothing convinces devs like another dev saying: “Yeah, I use this, and it actually works.”

**Trust is earned, not assumed**

Developers don’t believe marketing claims. They believe data, code, firsthand experience, and above all, other developers.

If you can’t back up your claims with proof, they’ll assume your product is just another overhyped letdown. And honestly? They’re probably right.

So if your strategy relies on clever messaging, polished landing pages, and vague performance claims, you’re already failing.

**The hard truth**

Marketing can get developers in the door, but it can’t make them stay. If your product doesn’t prove itself immediately, no amount of branding, storytelling, or thought leadership will save you. So stop treating developers like regular customers. They don’t care about your marketing. They care about what works. And if your tool actually works? Shut up and let them see for themselves.
